File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        r = [nums[-1]] * len(nums)

        for i in range(len(nums) - 2, -1, -1):
            r[i] = r[i+1] * nums[i]
        
        l = 1

        res = []
        for i in range(len(nums)):
            if i == 0: res.append(r[i+1])
            elif i == len(nums) - 1: res.append(l)
            else: res.append(l * r[i+1])
            l *= nums[i]
        return res

File: Data Structures & Algorithms/baseball-game/submission-1.cpp
class Solution {
public:
    int calPoints(vector<string>& operations) {
        int res = 0;
        stack<int> st;

        for (auto &o: operations) {
            if (o == "+") {
                int f = st.top();
                st.pop();

                int s = st.top();
                st.push(f);

                res += (f + s);
                st.push(f+s);
            } else if (o == "C") {
                res -= st.top();
                st.pop();

            } else if (o == "D") {
                int d = 2 * st.top();
                res += d;
                st.push(d);
            } else {
                int n = stoi(o);
                res += n;
                st.push(n);
            }

            if(!st.empty())
            cout << "top=" << st.top() << " ";
        }


        return res;
    }
};